### DISCUSSION TITLE
Is xplr support tree layout?

### DISCUSSION THREAD

**Discussion by Abdillah** [sayanarijit/xplr#638] [Q&A] [answered]
I'm searching for CLI software that support for collapse and expand directory and as configurable as xplr / broot.
Tree layout is very convenient and enhance when working with structured code repository.

Is xplr has the ability to expand and collapse directory without switching to new screen?

**Comment by sayanarijit** [comment]
xplr doesn't support tree view. Broot does, but I guess you want something different?

I can see how tree view might be more convenient when using a mouse, but I don't have a good intuition of how efficient the workflow would be without a mouse. Unless it's something like what broot does.

**Reply by Abdillah** [reply]
That's a POV alright.

I just imagining a tree view with hjkl move and tab to expand/collapse directory, and then some shortcut to search (recursive or non-recursive). I don't imagine a mouse navigation when submit above discussion.

Broot does have a "tree" but it looks cluttered when it is expanded by default. I ask for xplr because it looks tidy (if expand already possible).
Broot feature request is closed unfortunately: https://github.com/Canop/broot/issues/54.

**Comment by sayanarijit** [comment]
Hmm [this comment](https://github.com/Canop/broot/issues/54#issuecomment-1617891160) elaborates the intention well.

So, as I understand, the goal is to be able to always see the directory structure along with current location, and not necessary change the navigation workflow.

Sounds like an interesting concept to brainstorm on. I believe we can achieve something similar with an external plugin.

**Reply by Abdillah** [reply]
Wow.. xplr is awesome. 

That a usecase from me, thanks.

**Comment by sayanarijit** [comment, marked as answer]
Try this: https://github.com/sayanarijit/tree-view.xplr

Lots of scope for improvement I guess.

**Reply by Abdillah** [reply]
Thanks! I don't expect to be implemented so fast.
I will try to contribute something as I adapt this to my needs.
